### DISCUSSION TITLE
Upload file to Agent

### DISCUSSION THREAD

**Discussion by rafaelrsss** [OfficeDev/microsoft-365-agents-toolkit#12798] [Q&A] [answered]
My declarative agent is working fine, but I would like to attach/upload files. Why the upload button is disabled? Maybe because I need to publish?

I plan to instruct my agent on how to analyse an excel file depending on the file name or the content.. but I am unable to test as my attach button is not enabled.

Any clues?

**Comment by therealjohn** [comment, marked as answer]
That feature is planned to come soon. You'll be able to upload a file directly in the message box or embed a file in the package.
